lightning/src/ln/channelmonitor.rs

```diff
@@ -1630,127 +1461,40 @@ impl<ChanSigner: ChannelKeys> ChannelMonitor<ChanSigner> {
-								output: vec!(TxOut {
-									script_pubkey: self.destination_script.clone(),
-									value: htlc.amount_msat / 1000,
-								}),
-							};
-							let predicted_weight = single_htlc_tx.get_weight() + Self::get_witnesses_weight(&[if htlc.offered { InputDescriptors::RevokedOfferedHTLC } else { InputDescriptors::RevokedReceivedHTLC }]);
-							let height_timer = Self::get_height_timer(height, htlc.cltv_expiry);
-							let mut used_feerate;
-							if subtract_high_prio_fee!(self, fee_estimator, single_htlc_tx.output[0].value, predicted_weight, used_feerate) {
-								let sighash_parts = bip143::SighashComponents::new(&single_htlc_tx);
-								let (redeemscript, revocation_key) = sign_input!(sighash_parts, single_htlc_tx.input[0], Some(idx), htlc.amount_msat / 1000);
-								assert!(predicted_weight >= single_htlc_tx.get_weight());
-								log_trace!(self, "Outpoint {}:{} is being being claimed, if it doesn't succeed, a bumped claiming txn is going to be broadcast at height {}", single_htlc_tx.input[0].previous_output.txid, single_htlc_tx.input[0].previous_output.vout, height_timer);
-								let mut per_input_material = HashMap::with_capacity(1);
-								per_input_material.insert(single_htlc_tx.input[0].previous_output, InputMaterial::Revoked { script: redeemscript, pubkey: Some(revocation_pubkey), key: revocation_key, is_htlc: true, amount: htlc.amount_msat / 1000 });
-								match self.claimable_outpoints.entry(single_htlc_tx.input[0].previous_output) {
-									hash_map::Entry::Occupied(_) => {},
-									hash_map::Entry::Vacant(entry) => { entry.insert((single_htlc_tx.txid(), height)); }
-								}
-								match self.pending_claim_requests.entry(single_htlc_tx.txid()) {
-									hash_map::Entry::Occupied(_) => {},
-									hash_map::Entry::Vacant(entry) => { entry.insert(ClaimTxBumpMaterial { height_timer, feerate_previous: used_feerate, soonest_timelock: htlc.cltv_expiry, per_input_material }); }
-								}
-								txn_to_broadcast.push(single_htlc_tx);
-							}
+							return (claimable_outpoints, (commitment_txid, watch_outputs), spendable_descriptor); // Corrupted per_commitment_data, fuck this user
 						}
+						let witness_data = InputMaterial::Revoked { script: expected_script, pubkey: Some(revocation_pubkey), key: revocation_key, is_htlc: true, revoked_amount: tx.output[transaction_output_index as usize].value };
+						outpoints.push(ClaimRequest { absolute_timelock: htlc.cltv_expiry, aggregable: true, outpoint: BitcoinOutPoint { txid: commitment_txid, vout: transaction_output_index }, witness_data });
 					}
 				}
 			}
 
-			if !inputs.is_empty() || !txn_to_broadcast.is_empty() || per_commitment_option.is_some() { // ie we're confident this is actually ours
+			// Last, track onchain revoked commitment transaction and fail backward outgoing HTLCs as payment path is broken
+			if !claimable_outpoints.is_empty() || per_commitment_option.is_some() { // ie we're confident this is actually ours
```

Comment:
I dont think its possible for us to have inserted into claimable_outpoints by this point, only outpoints. I guess there are no tests which hit this?

Reply:
Hmmm is this check accurate even using outpoints, given per_commitment_option.is_some (and we shouldn't care about tracking revoked remote commitment tx on chain if there is no htlc outputs, which may explain why we don't hit with a test)